datanode/gateway/graphql/update_future_resolver.go

```diff
@@ -21,23 +21,25 @@ import (
 type updateFutureProductResolver VegaResolverRoot
 
 func (r *updateFutureProductResolver) DataSourceSpecForSettlementData(ctx context.Context, obj *vega.UpdateFutureProduct) (*DataSourceDefinition, error) {
-	if obj.DataSourceSpecForSettlementData != nil {
-		return nil, nil
+	ds := &DataSourceDefinition{}
+
+	if obj != nil {
```

Comment:
no need to check if `obj != nil`

Reply:
This and all below are addressed as well, thanks !